libraries/resource_providers/api500/c7000/volume_template_provider.rb

```diff
@@ -0,0 +1,51 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
```

Comment:
This whole library also feels a bit like reinveinting the wheel. I'd suggest looking into the `load_resource` method and rethinking/rewriting some of the methods.